zulipterminal/ui_tools/views.py

```diff
@@ -798,4 +798,176 @@ def __init__(self, controller: Any, msg: Any) -> None:
+                # each version contains previous version's content and subject
+                # timestamp of future version matches with
+                # current version's contents
+                timestamp = time.ctime(msg['edit_history'][version+1]
+                                          ['timestamp'])[:-5]
+                if 'prev_subject' in msg['edit_history'][version]:
+                    topic = [msg['edit_history'][version]['prev_subject']]
+                    orig_topic = topic
+                    if 'prev_subject' in msg['edit_history'][version+1]:
+                        if msg['edit_history'][version+1]['prev_subject'] !=\
+                           msg['edit_history'][version]['prev_subject']:
+                            topic.append(" (edited)")
+                else:
+                    topic = orig_topic
+                    if 'prev_subject' in msg['edit_history'][version+1]:
+                        topic.append(" (edited)")
+                if 'prev_content' in msg['edit_history'][version]:
+                    content = [msg['edit_history'][version]['prev_content']]
+                    orig_content = content
+                else:
+                    content = orig_content
+                edit_author_id = msg['edit_history'][version+1]['user_id']
+                self.edit_history_content(topic,
+                                          timestamp,
+                                          content,
+                                          edit_author_id)
+
+            # Original version
+            timestamp = time.ctime(msg['timestamp'])[:-5]
+            if 'prev_subject' in msg['edit_history'][-1]:
+                topic = [msg['edit_history'][-1]['prev_subject']]
+            else:
+                topic = orig_topic
+            if 'prev_content' in msg['edit_history'][-1]:
+                content = [msg['edit_history'][-1]['prev_content']]
+            else:
+                content = orig_content
+            self.edit_history_content(topic,
+                                      timestamp,
+                                      content,
+                                      self.msg['sender_id'])
+
+        # Unedited
+        else:
+            self.to_show.append(
+                urwid.Pile([
+                    (urwid.AttrWrap(
+                            urwid.Text("Unedited"), 'edit_history'))
+                    ]))
+            self.height += self.to_show[-1].rows((self.width,))
+
+        self.log = urwid.SimpleFocusListWalker(self.to_show)
+
+        super(EditHistoryView, self).__init__(self.log)
+
+    def edit_history_content(self,
+                             topic: List['str'],
+                             time: str,
+                             content: List[Any],
+                             edit_author_id: int) -> None:
+        # display who edited if edit_author not sender
+        if edit_author_id == self.msg['sender_id']:
+            self.to_show.append(
```

Comment:
Rather than appending to `self.to_show` from within the method, it may be clearer if this method returned the `Pile` to append?